src/validator/data_model.py

```
@@ -1050,3 +1057,156 @@ def _get_normal_vector(
+        if not v:
+            raise ValueError(f"Field '{info.field_name}' must not be empty.")
+        return v
+
+    @field_validator("maximum_flow_rate")
+    def validate_autosize_or_positive(cls, v: float | str) -> float | str:
+        if isinstance(v, str) and v.lower() == "autosize":
+            return "autosize"
+        try:
+            fv = float(v)
+            if fv <= 0:
+                raise ValueError("Value must be positive if specified numerically.")
```

Ruff (0.14.4)

Abstract `raise` to an inner function

(TRY301)